dev-tools/releng/src/main.rs

```diff
@@ -0,0 +1,669 @@
+            .await?;
+    }
+    // Record the branch and commit in the output
+    Command::new("git")
+        .arg("-C")
+        .arg(&args.helios_dir)
+        .args(["status", "--branch", "--porcelain=2"])
+        .ensure_success(&logger)
+        .await?;
+
+    // Check that the omicron1 brand is installed
+    if !Command::new("pkg")
+        .args(["verify", "-q", "/system/zones/brand/omicron1/tools"])
+        .is_success(&logger)
+        .await?
+    {
+        error!(
+            logger,
+            "the omicron1 brand is not installed; install it with \
+            `pfexec pkg install /system/zones/brand/omicron1/tools`"
+        );
+        preflight_ok = false;
+    }
+
+    // Check that the datasets for helios-image to use exist
+    for (dataset, option) in [
+        (&args.host_dataset, "--host-dataset"),
+        (&args.recovery_dataset, "--recovery-dataset"),
+    ] {
+        if !Command::new("zfs")
+            .arg("list")
+            .arg(dataset)
+            .is_success(&logger)
+            .await?
+        {
+            error!(
+                logger,
+                "the dataset {0} does not exist; run `pfexec zfs create \
+                -p {0}`, or specify a different one with {1}",
+                shell_words::quote(dataset),
+                option
+            );
+            preflight_ok = false;
+        }
+    }
+
+    if !preflight_ok {
+        bail!("some preflight checks failed");
+    }
+
+    fs::create_dir_all(&args.output_dir).await?;
+
+    // DEFINE JOBS ============================================================
+    let tempdir = camino_tempfile::tempdir()
+        .context("failed to create temporary directory")?;
+    let mut jobs = Jobs::new(&logger, permits.clone(), &args.output_dir);
+
+    jobs.push_command(
+        "helios-setup",
+        Command::new("ptime")
+            .args(["-m", "gmake", "setup"])
+            .current_dir(&args.helios_dir)
+            // ?!?!
+            // somehow, the Makefile does not see a new `$(PWD)` without this.
+            .env("PWD", &args.helios_dir)
+            // Setting `BUILD_OS` to no makes setup skip repositories we don't
+            // need for building the OS itself (we are just building an image
+            // from an already-built OS).
+            .env("BUILD_OS", "no"),
+    );
+
+    // Download the toolchain for phbl before we get to the image build steps.
+    // (This is possibly a micro-optimization.)
+    jobs.push_command(
+        "phbl-toolchain",
+        Command::new("cargo")
+            .arg("--version")
+            .current_dir(args.helios_dir.join("projects/phbl")),
+    )
+    .after("helios-setup");
+
+    jobs.push_command(
+        "omicron-package",
+        Command::new("ptime").args([
+            "-m",
+            "cargo",
+            "build",
+            "--locked",
+            "--release",
+            "--bin",
+            "omicron-package",
+        ]),
```

Comment:
Consider accepting an `OMICRON_PACKAGE` environment variable for when you want to bring your own binary.

Reply:
Is your thinking that setting `OMICRON_PACKAGE` would skip adding this job?

Reply:
Yeah. It can be pretty useful to copy a binary around without having to pay the price of building it.